src/components/dialog/index.js

```diff
@@ -49,210 +45,217 @@ class Dialog extends BaseComponent {
+     * this allows for scrollable content (the children of the dialog)
+     * props are transferred to the renderPannableHeader
      */
-    onModalDismissed: PropTypes.func
+    renderPannableHeader: PropTypes.func,
+    /**
+     * Migration flag, send true to use the new (and improved) Dialog, default is false
+     */
+    migrate: PropTypes.bool,
   };
 
   static defaultProps = {
+    migrate: false,
     overlayBackgroundColor: Colors.rgba(Colors.dark10, 0.6),
     width: '90%',
-    height: '70%',
-    useModal: true
   };
 
-  static swipeDirections = SWIPE_DIRECTIONS; // DEFRECATED
-
   constructor(props) {
     super(props);
 
-    this.initialPosition = props.top ? -Constants.screenHeight : Constants.screenHeight;
-
     this.state = {
       alignments: this.state.alignments,
-      deltaY: new Animated.Value(this.initialPosition)
+      orientationKey: undefined,
+      modalVisibility: props.visible,
+      dialogVisibility: props.visible,
     };
 
-    if (props.dismissSwipeDirection) {
-      console.warn('Dialog component\'s prop \'dismissSwipeDirection\' is deprecated, please remove it');
+    if (props.migrate) {
+      this.setAlignment();
     }
+  }
 
-    if (props.visible) {
-      this.animateContent();
-    }
+  componentDidMount() {
+    Constants.addDimensionsEventListener(this.onOrientationChange);
+  }
+
+  componentWillUnmount() {
+    Constants.removeDimensionsEventListener(this.onOrientationChange);
   }
 
   componentDidUpdate(prevProps) {
-    if (!prevProps.visible && this.props.visible) {
-      this.animateContent();
+    const {visible, top, bottom, width, height} = this.props;
+    const {visible: prevVisible, top: prevTop, bottom: prevBottom, width: prevWidth, height: prevHeight} = prevProps;
+
+    if (visible && !prevVisible) {
+      this.setState({modalVisibility: true, dialogVisibility: true});
+    } else if (prevVisible && !visible) {
+      this.hideDialogView();
     }
-  }
 
-  generateStyles() {
-    this.styles = createStyles(this.props);
-  }
+    if (top !== prevTop || bottom !== prevBottom) {
+      this.setAlignment();
+    }
 
-  onDismiss = () => {
-    this.initPositions();
-    _.invoke(this.props, 'onDismiss');
+    if (width !== prevWidth || height !== prevHeight) {
+      this.setSizeAndFlexType(width, height);
+    }
   }
 
-  animatedDismiss = () => {
-    if (this.panGestureViewRef) {
-      this.panGestureViewRef.animateDismiss();
-    } else {
-      this.onDismiss();
+  onOrientationChange = () => {
+    const orientationKey = Constants.orientation;
+    if (this.state.orientationKey !== orientationKey) {
+      this.setState({orientationKey});
     }
+  };
+
+  setSizeAndFlexType(width, height) {
+    this.styles.size = {width, height};
+    this.styles.flexType = height ? {flex: 1} : {flex: 0};
   }
 
-  initPositions() {
-    this.setState({
-      deltaY: new Animated.Value(this.initialPosition)
+  createStyles(props) {
```

Comment:
Why did u move this function into the class, not sure I follow what's going on here with all the styles.

Reply:
I thought it's more organized this way. Is there a drawback?

Reply:
Reverted (added a key in the screen as discussed)